Data from diet-induced obesity RNAseq atlas from 19 tissues have been deposited to GEO with accession number GSE185466. The analysis was performed in R (3.5 and 3.6) and python (2.7, 3.5 and 3.7). Libraries for data analysis include numpy, pandas, scikit-learn, scanpy and scipy. The datasets reported in this paper are available in supplementary materials. When indicated, network analysis and visualization were done using the python library NetworkX, and the software GEPHI. Functions to evaluate networks have been created and validated by the Ideker Lab (archived in https://github.com/idekerlab/Network_Evaluation_Tools) and are stored on the network data exchange (NDEx)170. Other functions to evaluate network topology can be found as inbuilt analysis tools in GEPHI. Tissue-specific networks for functional annotation of topological dependencies can be found at http://giant.princeton.edu and https://hb.flatironinstitute.org/. Positional gene enrichment analysis tool can be found at http://silico.biotoul.fr/pge/. Functional annotation tool from a large library of biological repositories can be found at https://maayanlab.cloud/Enrichr/.  DAVID functional clustering analysis tools can be found at https://david.ncifcrf.gov/. To filter out PGE regions without cytogenetic band enrichment, we used a curated list of cytogenetic band-gene dependencies archived in http://www.gsea-msigdb.org/gsea/msigdb/index.jsp. Prediction of TFBS from DNA sequences can be found at http://alggen.lsi.upc.es/. Prediction of conserved TFBS in promoter regions of genes of interest can be found at http://www.gsea-msigdb.org/gsea/msigdb/index.jsp. Tissue-naïve and tissue-specific protein-protein interactions used for n+1 network reconstruction can be found at http://www.interactome-atlas.org/. Tissue-naïve protein-protein interactions used for full network reconstruction can be found at  https://thebiogrid.org/. In silico prediction of intrinsically disordered regions in amino acid sequences can be found at http://www.pondr.com/, https://iupred3.elte.hu/, https://mobidb.bio.unipd.it/. Curated experimentally validated phase separation database stored at http://db.phasep.pro/. Context-dependent phase separation score from ensemble parameters of conformation changes present in disordered proteins. They offered discrepancies of conformations that are related to the local environment such as salt, solvation, multi-molecular interactions and ligand presence. These can be found at http://pappulab.wustl.edu/CIDER/about/ and can be implemented locally following instructions stored at http://pappulab.github.io/localCIDER/. Phylogenetic analysis tool information for protein residues conservation score and 3D modelling of residue exposure can be found at https://consurf.tau.ac.il/. Evolutionary domain conservation of protein domains with known structure can be found at https://consurfdb.tau.ac.il/. ChIP-atlas methodology, documentation and functions for analysis can be found at https://github.com/inutano/chip-atlas. Compendium of human chromatin contact maps has been created by the Ren Lab and is stored at GEO: GSE87112 (schmitt, bing ren 2016). Cis-regulatory elements relationship were obtained from GeneHancer, and can be found at http://www.genecards.org/. VISTA enhancer elements can be found at https://enhancer.lbl.gov/frnt_page_n.shtm. HACER, human active enhancer to interpret regulatory variants database is archived in http://bioinfo.vanderbilt.edu/AE/HACER/index.html. A curated database of cell-specific active regulatory regions can be found at http://www.enhanceratlas.org/index.php. Global chromatin 3D reconstruction tool and code can be found as follows: GSDB at http://sysbio.rnet.missouri.edu/3dgenome/GSDB/; and Spacewalk at http://3dg.io/spacewalk/. 3D-multiscale Monte Carlo modeling code for global and local CTCF-mediated ChIA-PET sequencing can be found at https://bitbucket.org/3dome/3dome_mmc/src/master/. Code for integration of structural variants (SVs) to spatial chromatin regions can be found at https://bitbucket.org/4dnucleome/spatial_chromatin_architecture. Data by Sadowski et al. on SVs spatial chromatin effects can be found at https://zenodo.org/record/2837248#.YCQBvZNKhPM. CTCF-binding site prediction tool can be found at https://insulatordb.uthsc.edu/.  Structural variants data from the reference resource compendia developed by Collins et al. 40 is archived in https://gnomad.broadinstitute.org/. To isolate metabolic disease associations for a given genomic-range, we used the curated common metabolic disease portal, archived in https://hugeamp.org/. For topological comparative genomics and to evaluate chromosomal synteny, tools can be found at the synteny portal, archived in http://bioinfo.konkuk.ac.kr/synteny_portal/. CAMMEL mediation analysis and polygenic risk scores can be found at https://github.com/YPARK/GTEx_mediation and https://github.com/YPARK/PRS.  We integrated in our pipelines previously published data derived from radial genome loci positioning by sequencing (GPSeq) method, which gives relative distances between genomic-loci and the nuclear lamina 59 (GEO:GSE135882; GPSeq protocol available at https://doi.org/10.21203/rs.3.pex-570/v1; data available at https://github.com/ggirelli/GPSeq-source-data). To evaluate radial positions of transcriptional active loci in single cells and their structural relationships, we used datasets from established genome-scale approaches that use multiplexed FISH imaging technology 60 (available at https://zenodo.org/record/3928890#.YDgeiZNKiuV with script functions at https://github.com/ZhuangLab/Chromatin_Analysis_2020_cell)
