Supplementary tables
Table S1. Metabolic phenotype comparison and human specific molecular signatures
Metabolic parameters and fasting endurance score in nonprimate mammals
Metabolic parameters and fasting endurance score in primate mammals
Metabolic parameters and fasting endurance score in hominoids
Tissue mass and metabolic parameters across mammals
GREAT analysis from ATACseq comparison between human-primate in fat: human increase 
GREAT analysis from ATACseq comparison between human-primate in fat: human decrease
Fat branch-specific acceleration and positive selection from ATACseq comparison between humans and primates
Accelerated evolution signature genes in brain
List of human accelerated regions coordinates
HAR-associated genes discovery from Capra et al. 2013
Full list of HAR-genes compiled for this study
Functional annotation of HAR genes in brain-specific networks
Functional annotation of HAR genes in fat-specific networks
Data

Table S2. Transcriptional regulatory networks of HAR-genes
ChIP-seq enrichment ChIP-atlas of HAR-genes
Adipocytes ChIP-seq enrichment ChIP-atlas of HAR-genes
Neural ChIP-seq enrichment ChIP-atlas of HAR-genes
Stem cells ChIP-seq enrichment ChIP-atlas of HAR-genes
Blood ChIP-seq enrichment ChIP-atlas of HAR-genes
Muscle ChIP-seq enrichment ChIP-atlas of HAR-genes
Liver ChIP-seq enrichment ChIP-atlas of HAR-genes
TFBS enrichment from MSDB
TFBS enrichment form genome browser
ENCODE and ChEA consensus enrichment
Data

Table S3. Transcriptional regulatory networks, and PPI reconstitution
Node + 1 network reconstruction of PPIs for HAR-genes regulators
Network topological prioritization of network in A
Full PGC1A-network reconstruction of protein-protein interactions
Network topological prioritization of full PGC1A-network
Reactome functional annotation of HAR cooperative regulators 
Gene-ontology functional annotation of HAR cooperative regulators
Fat and brain tissue-specific PPIs for PGC1A-network 
Data 

Table S4. Transcriptional regulation of HAR-genes and metabolic-HAR hubs 
Functionally-associated and interacting transcriptional regulators: co-regulated genes by promoter ChIP-seq enrichment
Co-bound transcriptional regulators: co-regulated HAR-genes (metabolic-HAR genes) 
Positional gene enrichments of metabolic-HAR genes
Cytogenetic band enrichments of metabolic-HAR genes
Functional annotation of metabolic-HAR PGE-domains 
Data

Table S5. Compendia of HiC data and hESC interacting interchromosomal regions
HiC chromatin interacting regions
Frequently interacting regions (FIREs)
Network from hESC-HiC interacting chromatin segments and metabolic-HAR regions overlap
Network ID of interacting segments in A
Topological prioritization and network analysis of hESC-HiC interacting chromatin segments
Data

Table S6. Genome-wide network of enhancer-promoter relationships
GeneHancer database of enhancer-promoter associations.
Regulatory elements associations
Network analyses of regulatory elements associations
Module-based and degree prioritization
Degree-based prioritization of enhancers
Social promoter genes and genomic cluster positions
Nonsocial promoter genes and genomic cluster positions
Data

Table S7. mHAR genomic hubs subnetworks from linked transcriptional compartments
ATG4C-submodule short-range relationships
DGAT1-submodule short-range relationships
PGC1A-submodule short-range relationships
CAT-submodule short-range relationships
ATG4C-submodule territorial ChIP-seq enrichment for social enhancer and promoters
DGAT-submodule territorial ChIP-seq enrichment for social enhancer and promoters
PGC1A-submodule territorial ChIP-seq enrichment for social enhancer and promoters
CAT-submodule short-range relationships
PPI network reconstruction of territorial ChIP-seq enrichment of regulators within ATG4C submodule
Data

Table S8. Genomic range genetics of metabolic-HAR compartments
ATG4C metabolic-HAR domain genomic range genetics (chr1:62mb)
INSIG2 metabolic-HAR domain genomic range genetics  (chr2:110mb)
PGC1A metabolic-HAR domain genomic range genetics  (chr4:25mb)
CAT metabolic-HAR domain genomic range genetics  (chr8:38mb)
APOA5 metabolic-HAR domain genomic range genetics  (chr11:117mb)
Data

Table S9. Synteny of metabolic-HAR regions and mammalian perturbation networks
Murine transcriptional hubs of metabolic-HAR regions
Metabolic-HAR gene perturbation-phenotype association
Gene network topological dependencies
Phenotype network topological dependencies
Data

Table S10. ChIP-MS structuring candidates and PGC1A-network 
List of proteins identified by chromatin proteomic profiling in mESC dual enhancer-promoter
Interrogation of PPIs for PGC1A-network and ChIP-MS structuring candidates
IDs node summary annotation for B
Functional network annotation 
Network topological dependencies
Data

Table S11. Intrinsically disordered regions and conservation of HAR-gene regulator
Consensus disordered regions in PGC1A protein
Hydropathy boundary ordered and disordered proteins
Multivalent pi-pi disordered score human proteins 
DisProt disordered score human proteins 
Consurf amino acid variation of PGC1A residues
Amino acid conservation scores for PGC1A co-activator
Data

Table S12. Summary table of published datasets used for integration and analysis
Link
