
Supplementary Tables

Table S1. Metabolic phenotype comparison across mammals 
Metabolic parameters and fasting endurance score in nonprimate mammals
Metabolic parameters and fasting endurance score in primate mammals
Metabolic parameters and fasting endurance score in hominoids
Tissue mass and metabolic parameters across mammals
Link: https://www.dropbox.com/s/7czlr2dkukutsxl/Table%20S1.xlsx?dl=0 

Table S2. Differential human specific molecular signatures in fat and brain
GREAT analysis from ATACseq comparison between human-primate in fat: human increase 
GREAT analysis from ATACseq comparison between human-primate in fat: human decrease
Fat branch-specific acceleration and positive selection from ATACseq comparison between humans and primates
Accelerated evolution signature genes in brain
Link: https://www.dropbox.com/s/h0bva3hjg0jx37c/Table%20S2.xlsx?dl=0 

Table S3. Human-accelerated regulatory regions
List of human accelerated regions coordinates
HAR-associated genes discovery from Capra et al. 2013
Full list of HAR-genes compiled for this study
Functional annotation of HAR genes in brain-specific networks
Functional annotation of HAR genes in fat-specific networks
Link: https://www.dropbox.com/s/j89lhmskpxoo8x7/Table%20S3.xlsx?dl=0 

Table S4. Transcriptional regulatory networks of HAR-genes
ChIP-seq enrichment ChIP-atlas of HAR-genes
Adipocytes ChIP-seq enrichment ChIP-atlas of HAR-genes
Neural ChIP-seq enrichment ChIP-atlas of HAR-genes
Stem cells ChIP-seq enrichment ChIP-atlas of HAR-genes
Blood ChIP-seq enrichment ChIP-atlas of HAR-genes
Muscle ChIP-seq enrichment ChIP-atlas of HAR-genes
Liver ChIP-seq enrichment ChIP-atlas of HAR-genes
TFBS enrichment from MSDB
TFBS enrichment form genome browser
ENCODE and ChEA consensus enrichment
https://www.dropbox.com/s/1sgdr5741pqg6t0/Table%20S4.xlsx?dl=0 

Table S5. Transcriptional regulatory networks, and PPI reconstitution
Node + 1 network reconstruction of PPIs for HAR-genes regulators
Network topological prioritization of network in A
Full PGC1A-network reconstruction of protein-protein interactions
Network topological prioritization of full PGC1A-network
Reactome functional annotation of HAR cooperative regulators 
Gene-ontology functional annotation of HAR cooperative regulators
Fat and brain tissue-specific PPIs for PGC1A-network  
https://www.dropbox.com/s/brp2olmem6o5zgy/Table%20S5.xlsx?dl=0 

Table S6. Combinatorial transcriptional regulation of HAR-genes
Functional interacting transcriptional regulators co-regulated genes by promoter ChIP-seq enrichment
Combinatorial co-regulated HAR-genes (metabolic-HAR genes) 
https://www.dropbox.com/s/0zq6kakrtw5q773/Table%20S8.xlsx?dl=0 

Table S7. Functional topological loci decomposition of metabolic-HAR regions 
Positional gene enrichments of metabolic-HAR genes
Cytogenetic band enrichments of metabolic-HAR genes
Functional annotation of metabolic-HAR PGE-domains 
https://www.dropbox.com/s/10wlpjohnu4n83u/Table%20S9.xlsx?dl=0 

Table S8. Compendia of HiC data
HiC chromatin interacting regions
FIREs
https://www.dropbox.com/s/uejs9r1h405i5n2/Table%20S10.xlsx?dl=0 

Table S9. hESC interacting interchromosomal regions
Network from hESC-HiC interacting chromatin segments and metabolic-HAR regions overlap
Network ID of interacting segments in A
Topological prioritization and network analysis of hESC-HiC interacting chromatin segments
https://www.dropbox.com/s/4hl6jvb7l87enbb/Table%20S11.xlsx?dl=0 

Table S10. Genome-wide short-range enhancer-promoter relationships
GeneHancer database of enhancer-promoter associations.
https://www.dropbox.com/s/ih9mof2smzyc6os/Table%20S12.xlsx?dl=0 

Table S11.  Genome-wide short-range regulatory elements network
Regulatory elements associations
Network analyses of regulatory elements associations
Module-based and degree prioritization
Degree-based prioritization of enhancers
Social promoter genes and genomic cluster positions
Nonsocial promoter genes and genomic cluster positions
https://www.dropbox.com/s/4b4u8hmns6u3el9/Table%20S13.xlsx?dl=0 

Table S12. metabolic-HAR domains short-range regulatory subnetwork from reconstituted nuclear compartment 
ATG4C-submodule short-range relationships
DGAT1-submodule short-range relationships
PGC1A-submodule short-range relationships
CAT-submodule short-range relationships
ATG4C-submodule territorial ChIP-seq enrichment for social enhancer and promoters
DGAT-submodule territorial ChIP-seq enrichment for social enhancer and promoters
PGC1A-submodule territorial ChIP-seq enrichment for social enhancer and promoters
CAT-submodule short-range relationships
PPI network reconstruction of territorial ChIP-seq enrichment of regulators within ATG4C submodule
https://www.dropbox.com/s/dcy05kn2hjbcaz2/Table%20S14.xlsx?dl=0 

Table S13. Active enhancer elements
Vista active enhancer elements human and mouse coordinates and associated genes
Enhancer identity and human coordinates
https://www.dropbox.com/s/2tpw0kos0xqyndv/Table%20S15.xlsx?dl=0 

Table S14. Structural variants, genomic range genetics of metabolic-HAR compartments
Common structural variants with SNVs
Number of SVs in reconstituted nuclear compartment with metabolic-HAR domains 
ATG4C metabolic-HAR domain genomic range genetics (chr1:62mb)
INSIG2 metabolic-HAR domain genomic range genetics  (chr2:110mb)
PGC1A metabolic-HAR domain genomic range genetics  (chr4:25mb)
CAT metabolic-HAR domain genomic range genetics  (chr8:38mb)
APOA5 metabolic-HAR domain genomic range genetics  (chr11:117mb)
GREAT analysis metabolic-HAR genomic coordinates
https://www.dropbox.com/s/gun54eshm93xl3u/Table%20S16.xlsx?dl=0 

Table S15. Synteny of metabolic-HAR regions and mammalian perturbation networks
Murine transcriptional hubs of metabolic-HAR regions
Metabolic-HAR gene perturbation-phenotype association
Gene network topological dependencies
Phenotype network topological dependencies
https://www.dropbox.com/s/7aq1nhhnwsbp5fq/Table%20S22.xlsx?dl=0 

Table S16. Differentially expressed genes in diet-induced obesity across 19 organs
Mesenteric adipose tissue. B. Brown adipose tissue. C. Visceral adipose tissue. D. Retroperitoneal adipose tissue. E. Subcutaneous adipose tissue. F. Hypothalamus. G. Ileum. H. Jejunum. I. Kidney. J. Liver. K. Stomach. L. Duodenum. M. Soleus. N. Colon. O. Cortex. P. Adrenal gland. Q. Cerebellum. R. Midbrain stem.   
https://www.dropbox.com/s/lmd3k8k8xaiveth/Table%20S23.xlsx?dl=0 

Table S17. Transcriptional hubs controlled by diet 
Mesenteric adipose tissue. B. Brown adipose tissue. C. Visceral adipose tissue. D. Retroperitoneal adipose tissue. E. Subcutaneous adipose tissue. F. Hypothalamus. G. Ileum. H. Jejunum. I. Kidney. J. Liver. K. Stomach. L. Duodenum. M. Soleus. N. Colon. O. Cortex. P. Adrenal gland. Q. Cerebellum. R. Midbrain stem. 
https://www.dropbox.com/s/8e2w8wrzxtvxsws/Table%20S24.xlsx?dl=0 

Table S18. Fat-specific transcriptional hubs  
Fat-specific transcriptional hubs controlled by diet
https://www.dropbox.com/s/bq1fe09wox9krkr/Table%20S25.xlsx?dl=0 

Table S19. Single cell topological decomposition of fat transcriptional hubs
Cell-specific genes in murine adipose from Tabula Muris consortium
Natural killer cell-specific transcriptional hubs in murine adipose tissue
T cell-specific transcriptional hubs in murine adipose tissue
Myeloid cell-specific transcriptional hubs in murine adipose tissue
B cell-specific transcriptional hubs in murine adipose tissue
Mesenchymal cell-specific transcriptional hubs in murine adipose tissue
Endothelial cell-specific transcriptional hubs in murine adipose tissue
https://www.dropbox.com/s/oesjvj972mu0dna/Table%20S26.xlsx?dl=0 

Table S20. ChIP-MS structuring candidates and PGC1A-network 
List of proteins identified by chromatin proteomic profiling in mESC dual enhancer-promoter
Interrogation of PPIs for PGC1A-network and ChIP-MS structuring candidates
IDs node summary annotation for B
Functional network annotation 
Network topological dependencies
https://www.dropbox.com/s/c8helx3ck1engfa/Table%20S18.xlsx?dl=0 

Table S21. Intrinsically disordered regions of HAR-gene regulators
Consensus disordered regions in PGC1A protein
Hydropathy boundary ordered and disordered proteins
Multivalent pi-pi disordered score human proteins 
DisProt disordered score human proteins 
https://www.dropbox.com/s/idqxtlylhl1lg24/Table%20S6.xlsx?dl=0 

Table S22. Phylogenetic conservation of co-activator protein residues
Consurf amino acid variation of PGC1A residues
Amino acid conservation scores for PGC1A co-activator
https://www.dropbox.com/s/57k3ql68793bwql/Table%20S7.xlsx?dl=0

Table S23. Tissue- and cell-specific enhancer activity
Tissue- and cell-specific enhancer activity 
Global mean and sum of activity for each enhancer coordinate
Adipocyte-specific enhancer
https://www.dropbox.com/s/10ojb9escn0h9zm/Table%20S19.xlsx?dl=0 

Table S24. Human active enhancers from GRO-seq, PRO-seq and CAGE data https://www.dropbox.com/s/whsch6czar35mrz/Table%20S20.txt?dl=0 

Table S25. Radial chromatin organization
GPseq radial chromatin loci coordinates
DNA-MERFISH chromatin organization
Ontology of core and nuclear speckles associated genes with transcriptional burst on from B
Biological enrichment of core and nuclear speckles associated genes with transcriptional burst on
https://www.dropbox.com/s/n6iayzilav1x8if/Table%20S28.xlsx?dl=0 

Table S26. Radial transcriptional regulators  
TFBS density and GPseq score radial correlations
Functional interacting regulators with peripheral enrichment
Node + 1 network reconstruction of B
Functional interacting regulators with core enrichment
Node + 1 network reconstruction of D
https://www.dropbox.com/s/ymx5kshsev9kpn1/Table%20S29.xlsx?dl=0 

Table S27. Radial transcriptional condensate features 
Amino acid sequences for core enriched regulators
Amino acid sequences for peripherally enriched regulators
Context-dependent condensate features for core enriched regulators
Context-dependent condensate features for peripherally enriched regulators
https://www.dropbox.com/s/v69f5yb6p7ap73y/Table%20S30.xlsx?dl=0 

Table S28. Multi-tissue mediators of polygenic risk scores for BMI and T2D
Tissue-specific mediators of polygenic risk for T2D (BMI-adjusted) 
Tissue-specific mediators of polygenic risk for BMI (GIANT)
Genomic hubs for BMI mediators of polygenic risk
Genomic hubs for T2D mediators of polygenic risk
https://www.dropbox.com/s/efr9d3oq1ihikh4/Table%20S27.xlsx?dl=0 

Table S29. Social interchromosomal network: interacting metabolic disease hubs and metabolic-HAR regions  
Social interchromosomal regions: 5mb range interacting scores
Interacting metabolic disease hubs for shared hotspots for BMI and T2D polygenic risk
Interacting metabolic disease hubs coordinates for BMI mediators of polygenic risk
Interacting metabolic disease hubs coordinates for T2D mediators of polygenic risk
Mammalian phenotype integration of gene perturbations for shared mediators of BMI and T2D polygenic risk
Network topological dependencies for mediators
Network topological dependencies for phenotypes
https://www.dropbox.com/s/47fnm8cqlijs5ki/Table%20S31.xlsx?dl=0 

Table S30. Interacting metabolic disease hubs regulated by PGC1A-network 
Combinatorial regulation of T2D hotspots by PGC1A-network
Combinatorial regulation of BMI hotspots by PGC1A-network
Gene list > 4 combinatorial PGC1A-network for T2D-BMI mediators of polygenic risk
Genomic clusters by cytogenetic band enrichments
Genomic clusters by positional gene enrichments
https://www.dropbox.com/s/0b95i0r8x3ng4n4/Table%20S32.xlsx?dl=0 

Table S31. Single cell topological decomposition of interacting metabolic disease hubs 
Subcutaneous progenitors genes in obese fat scRNAseq
Visceral progenitors genes in obese fat scRNA
Immune cell-specific genes in obese fat scRNAseq 
Endothelial cell-specific genes in obese fat scRNAseq
Genomic hubs from fat progenitors scRNAseq genes
Genomic hubs from fat immune scRNAseq genes
https://www.dropbox.com/s/ml1zqsonmgezt6h/Table%20S33.xlsx?dl=0 
