1. # **Supplementary tables**

   1. ## **Table S1. Metabolic phenotype comparison and human specific molecular signatures**

      1. Metabolic parameters and fasting endurance score in nonprimate mammals  
      2. Metabolic parameters and fasting endurance score in primate mammals  
      3. Metabolic parameters and fasting endurance score in hominoids  
      4. Tissue mass and metabolic parameters across mammals  
      5. GREAT analysis from ATACseq comparison between human-primate in fat: human increase   
      6. GREAT analysis from ATACseq comparison between human-primate in fat: human decrease  
      7. Fat branch-specific acceleration and positive selection from ATACseq comparison between humans and primates  
      8. Accelerated evolution signature genes in brain  
      9. List of human accelerated regions coordinates  
      10. HAR-associated genes discovery from Capra et al. 2013  
      11. Full list of HAR-genes compiled for this study  
      12. Functional annotation of HAR genes in brain-specific networks  
      13. Functional annotation of HAR genes in fat-specific networks  
      14. [Data](https://www.dropbox.com/scl/fi/w8rn4o5j8kdh4u526it86/Table-S1.xlsx?rlkey=6gzvv31zns96vmmsskr9i5g4l&dl=0)  
          

   2. ## **Table S2. Transcriptional regulatory networks of HAR-genes**

      1. ChIP-seq enrichment ChIP-atlas of HAR-genes  
      2. Adipocytes ChIP-seq enrichment ChIP-atlas of HAR-genes  
      3. Neural ChIP-seq enrichment ChIP-atlas of HAR-genes  
      4. Stem cells ChIP-seq enrichment ChIP-atlas of HAR-genes  
      5. Blood ChIP-seq enrichment ChIP-atlas of HAR-genes  
      6. Muscle ChIP-seq enrichment ChIP-atlas of HAR-genes  
      7. Liver ChIP-seq enrichment ChIP-atlas of HAR-genes  
      8. TFBS enrichment from MSDB  
      9. TFBS enrichment form genome browser  
      10. ENCODE and ChEA consensus enrichment  
      11. [Data](https://www.dropbox.com/scl/fi/kxxgjwfz0zvii27uip0cj/Table-S2.xlsx?rlkey=rwdao1qhjfrzv0fhaqg82kw9l&dl=0)

   3. ## **Table S3. Transcriptional regulatory networks, and PPI reconstitution**

      1. Node \+ 1 network reconstruction of PPIs for HAR-genes regulators  
      2. Network topological prioritization of network in A  
      3. Full PGC1A-network reconstruction of protein-protein interactions  
      4. Network topological prioritization of full PGC1A-network  
      5. Reactome functional annotation of HAR cooperative regulators   
      6. Gene-ontology functional annotation of HAR cooperative regulators  
      7. Fat and brain tissue-specific PPIs for PGC1A-network   
      8. [Data](https://www.dropbox.com/scl/fi/q962grygckzop6mv1ihvg/Table-S3.xlsx?rlkey=wm0l3jqplbb0rjl5snaypi6oi&dl=0)   
         

   4. ## **Table S4. Transcriptional regulation of HAR-genes and metabolic-HAR hubs** 

      1. Functionally-associated and interacting transcriptional regulators: co-regulated genes by promoter ChIP-seq enrichment  
      2. Co-bound transcriptional regulators: co-regulated HAR-genes (metabolic-HAR genes)   
      3. Positional gene enrichments of metabolic-HAR genes  
      4. Cytogenetic band enrichments of metabolic-HAR genes  
      5. Functional annotation of metabolic-HAR PGE-domains   
      6. [Data](https://www.dropbox.com/scl/fi/slt135n78g4lpzqic1ka5/Table-S4.xlsx?rlkey=eqkk62gem1cnbh1jkkqvf0eih&dl=0)  
         

   5. ## **Table S5. Compendia of HiC data and hESC interacting interchromosomal regions**

      1. HiC chromatin interacting regions  
      2. Frequently interacting regions (FIREs)  
      3. Network from hESC-HiC interacting chromatin segments and metabolic-HAR regions overlap  
      4. Network ID of interacting segments in A  
      5. Topological prioritization and network analysis of hESC-HiC interacting chromatin segments  
      6. [Data](https://www.dropbox.com/scl/fi/cxrioinbbl3lhkt2buait/Table-S5.xlsx?rlkey=npyor928g2bz026cxwfosjg01&dl=0)  
         

   6. ## **Table S6. Genome-wide network of enhancer-promoter relationships**

      1. GeneHancer database of enhancer-promoter associations.  
      2. Regulatory elements associations  
      3. Network analyses of regulatory elements associations  
      4. Module-based and degree prioritization  
      5. Degree-based prioritization of enhancers  
      6. Social promoter genes and genomic cluster positions  
      7. Nonsocial promoter genes and genomic cluster positions  
      8. [Data](https://www.dropbox.com/scl/fi/g08clf1gxky9a71fzjjxm/Table-S6.xlsx?rlkey=iw9vjzun3br0k89lra1be9txu&dl=0)  
         

   7. ## **Table S7. mHAR genomic hubs subnetworks from linked transcriptional compartments**

      1. ATG4C-submodule short-range relationships  
      2. DGAT1-submodule short-range relationships  
      3. PGC1A-submodule short-range relationships  
      4. CAT-submodule short-range relationships  
      5. ATG4C-submodule territorial ChIP-seq enrichment for social enhancer and promoters  
      6. DGAT-submodule territorial ChIP-seq enrichment for social enhancer and promoters  
      7. PGC1A-submodule territorial ChIP-seq enrichment for social enhancer and promoters  
      8. CAT-submodule short-range relationships  
      9. PPI network reconstruction of territorial ChIP-seq enrichment of regulators within ATG4C submodule  
      10. [Data](https://www.dropbox.com/scl/fi/vim16rgd41i3ripr3wwzl/Table-S7.xlsx?rlkey=vzo29w5yvestb2y724ll6fyx9&dl=0)  
          

   8. ## **Table S8. Genomic range genetics of metabolic-HAR compartments**

      1. ATG4C metabolic-HAR domain genomic range genetics (chr1:62mb)  
      2. INSIG2 metabolic-HAR domain genomic range genetics  (chr2:110mb)  
      3. PGC1A metabolic-HAR domain genomic range genetics  (chr4:25mb)  
      4. CAT metabolic-HAR domain genomic range genetics  (chr8:38mb)  
      5. APOA5 metabolic-HAR domain genomic range genetics  (chr11:117mb)  
      6. [Data](https://www.dropbox.com/scl/fi/nxyjqjvtmv44s5x7d7t9m/Table-S8.xlsx?rlkey=h9ldmma2uk0dajvaodt9aur28&dl=0)  
         

   9. ## **Table S9. Synteny of metabolic-HAR regions and mammalian perturbation networks**

      1. Murine transcriptional hubs of metabolic-HAR regions  
      2. Metabolic-HAR gene perturbation-phenotype association  
      3. Gene network topological dependencies  
      4. Phenotype network topological dependencies  
      5. [Data](https://www.dropbox.com/scl/fi/qft8p0ucrsxlt200vltt3/Table-S9.xlsx?rlkey=qof2vwzv3575gyd0tkp0l95ai&dl=0)  
         

   10. ## **Table S10. ChIP-MS structuring candidates and PGC1A-network** 

       1. List of proteins identified by chromatin proteomic profiling in mESC dual enhancer-promoter  
       2. Interrogation of PPIs for PGC1A-network and ChIP-MS structuring candidates  
       3. IDs node summary annotation for B  
       4. Functional network annotation   
       5. Network topological dependencies  
       6. [Data](https://www.dropbox.com/scl/fi/qft8p0ucrsxlt200vltt3/Table-S9.xlsx?rlkey=qof2vwzv3575gyd0tkp0l95ai&dl=0)  
          

   11. ## **Table S11. Intrinsically disordered regions and conservation of HAR-gene regulator**

       1. Consensus disordered regions in PGC1A protein  
       2. Hydropathy boundary ordered and disordered proteins  
       3. Multivalent pi-pi disordered score human proteins   
       4. DisProt disordered score human proteins   
       5. Consurf amino acid variation of PGC1A residues  
       6. Amino acid conservation scores for PGC1A co-activator  
       7. [Data](https://www.dropbox.com/scl/fi/i9iqvuh811zd6v0iq9xqt/Table-S11.xlsx?rlkey=ez84nk0cft6o0o24h031da48t&dl=0)  
          

   12. ## **Table S12. Summary table of published datasets used for integration and analysis**

       1. [Link](https://www.dropbox.com/scl/fi/9se1svuzlxhnxcpq96dct/TABLE-S12.xlsx?rlkey=oqgyddjcq5g8ycn9tcspcq24b&dl=0)
